### PR TITLE
TINY-11377: Converted `isEmpty` to an assertion function, improving cases with optional object properties

### PR DESCRIPTION
**Pull request by TheSpyder**
Related Ticket: TINY-11377

Description of Changes:
* Not actually related to the ticket, something I came across while working on it. The code in the test won't compile without the assertion return type and it's a pattern I was trying to use.

Pre-checks:
* [x] ~Changelog entry added~
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [ ] Milestone set
* [x] ~Docs ticket created (if applicable)~

## Summary by CodeRabbit

- **New Features**
  - Introduced a new function to create potentially empty objects for testing.
  
- **Bug Fixes**
  - Enhanced type safety of the `isEmpty` function, ensuring accurate type inference for empty objects.

- **Tests**
  - Expanded the test suite for `isEmpty` with additional cases to validate TypeScript's understanding of empty object types.

### CODE REVIEW

File: modules/katamari/src/test/ts/atomic/api/obj/ObjIsEmptyTest.ts [hunk 1]
```diff
@@ -20,4 +22,12 @@ describe('atomic.katamari.api.obj.ObjIsEmptyTest', () => {
       }
     ));
   });
+
+  it('helps TypeScript understand the empty type', () => {
```
Comment (spocke) [inline modules/katamari/src/test/ts/atomic/api/obj/ObjIsEmptyTest.ts:26]: Could we also add some form of tests that checks when it's empty. Like `if (isEmpty({})) { assert.deepEqual({}, {}) }` or something os that it checks that result type became an empty object.

**Review by spocke — APPROVED** [review]

File: modules/katamari/src/test/ts/atomic/api/obj/ObjIsEmptyTest.ts (hunk 1 above)
Comment (TheSpyder) [reply modules/katamari/src/test/ts/atomic/api/obj/ObjIsEmptyTest.ts:26]: TypeScript allows assigning any values to a `{}`	type, so it's mostly a pointless test, but sure.

**Review by kimwoodfield — APPROVED** [review]

**Comment by TheSpyder** [comment]
Test failure appears to be an unrelated flake, at least it was on `main`
